app/src/main/kotlin/com/wire/android/WireApplication.kt

```diff
@@ -121,6 +127,19 @@ class WireApplication : BaseApp() {
 
             appLogger.i("$TAG global observers")
             globalObserversManager.get().observe()
+
+            observeRecentlyEndedCall()
+        }
+    }
+
+    private suspend fun observeRecentlyEndedCall() {
+        coreLogic.get().getGlobalScope().session.currentSessionFlow().collectLatest { sessionResult ->
+            if (sessionResult is CurrentSessionResult.Success && sessionResult.accountInfo.isValid()) {
+                coreLogic.get().getSessionScope(sessionResult.accountInfo.userId).calls.observeRecentlyEndedCallMetadata()
+                    .collect { metadata ->
```

Comment:
I don't remember why exactly, but i remember we should avoid `collect{ flow.collect { } }` pattern. Much better to use `flatMap`, `flatMapLatest`...

Reply:
🤔 could you try to get some more context on that? The logic that happens in the `collect` is non suspending, so I believe it doesnt make difference here, but there might be something that I miss :D

Reply:
ah I got you now, you are right! changing it up

Reply:
@borichellow you can check it now :)